File: demo-material3/lib/features/peer_evaluation/data/datasources/local/local_assessment_source.dart
import '../../../domain/models/assessment.dart';
import '../../../domain/models/course.dart';
import '../../../domain/models/criteria.dart';
import '../../../domain/models/peer.dart';
import '../i_assessment_source.dart';

class LocalAssessmentSource implements IAssessmentSource {
  @override
  List<Criteria> getCriteria() => const [
        Criteria(id: 'crit-1', label: 'Puntualidad',    description: 'Llega puntual a las reuniones del equipo',                     weight: 0.25),
        Criteria(id: 'crit-2', label: 'Contribuciones', description: 'Aporta ideas y trabajo de calidad al proyecto',                 weight: 0.30),
        Criteria(id: 'crit-3', label: 'Compromiso',     description: 'Cumple con los entregables acordados en el tiempo establecido', weight: 0.25),
        Criteria(id: 'crit-4', label: 'Actitud',        description: 'Mantiene una actitud positiva y constructiva',                  weight: 0.20),
      ];

  @override
  List<Course> getCourses() => [
        Course(
          id: 'course-1',
          name: 'Desarrollo de Software',
          semester: '2025-1',
          enrollmentCode: 'DS-2025A',
          studentCount: 28,
        ),
        Course(
          id: 'course-2',
          name: 'Ingeniería de Requisitos',
          semester: '2025-1',
          enrollmentCode: 'IR-2025A',
          studentCount: 24,
        ),
        Course(
          id: 'course-3',
          name: 'Arquitecturas de Software',
          semester: '2025-1',
          enrollmentCode: 'AS-2025A',
          studentCount: 20,
        ),
      ];

  @override
  List<Assessment> getAssessments() => [
        Assessment(
          id: 'assess-1',
          courseId: 'course-1',
          title: 'Evaluación Parcial — Sprint 2',
          deadline: DateTime(2025, 4, 15),
          criteriaIds: ['crit-1', 'crit-2', 'crit-3', 'crit-4'],
          peerIds: ['peer-1', 'peer-2', 'peer-3', 'peer-4'],
        ),
        Assessment(
          id: 'assess-2',
          courseId: 'course-1',
          title: 'Evaluación Final — Sprint 4',
          deadline: DateTime(2025, 6, 10),
          criteriaIds: ['crit-1', 'crit-2', 'crit-3', 'crit-4'],
          peerIds: ['peer-1', 'peer-2', 'peer-3', 'peer-4'],
        ),
        Assessment(
          id: 'assess-3',
          courseId: 'course-2',
          title: 'Evaluación Midterm',
          deadline: DateTime(2025, 5, 1),
          criteriaIds: ['crit-1', 'crit-2', 'crit-3', 'crit-4'],
          peerIds: ['peer-5', 'peer-6'],
        ),
      ];

  @override
  List<Peer> getPeers() => const [
        Peer(id: 'peer-1', name: 'María García',  initials: 'MG'),
        Peer(id: 'peer-2', name: 'Carlos López',  initials: 'CL'),
        Peer(id: 'peer-3', name: 'Ana Martínez',  initials: 'AM'),
        Peer(id: 'peer-4', name: 'Julián Torres', initials: 'JT'),
        Peer(id: 'peer-5', name: 'Laura Ríos',    initials: 'LR'),
        Peer(id: 'peer-6', name: 'Diego Vargas',  initials: 'DV'),
      ];
}

File: demo-liquid-glass/DomainKit/Sources/DomainKit/data/datasources/local/LocalAssessmentSource.swift
// data/datasources/local/LocalAssessmentSource.swift
// Provides hardcoded demo data — simulates what a real network/database source would return

public final class LocalAssessmentSource {
    public init() {}

    public func fetchCourses() -> [Course] {
        [
            Course(
                id: "course-1",
                name: "Desarrollo de Software",
                semester: "2025-1",
                enrollmentCode: "DS-2025A",
                studentCount: 28
            ),
            Course(
                id: "course-2",
                name: "Ingeniería de Requisitos",
                semester: "2025-1",
                enrollmentCode: "IR-2025A",
                studentCount: 24
            ),
            Course(
                id: "course-3",
                name: "Arquitecturas de Software",
                semester: "2025-1",
                enrollmentCode: "AS-2025A",
                studentCount: 20
            ),
        ]
    }

    public func fetchAssessments() -> [Assessment] {
        [
            Assessment(
                id: "assess-1",
                title: "Evaluación Parcial — Sprint 2",
                courseId: "course-1",
                deadline: "2025-04-15",
                criteriaIds: ["crit-1", "crit-2", "crit-3", "crit-4"],
                peerIds: ["peer-1", "peer-2", "peer-3", "peer-4"]
            ),
            Assessment(
                id: "assess-2",
                title: "Evaluación Final — Sprint 4",
                courseId: "course-1",
                deadline: "2025-06-10",
                criteriaIds: ["crit-1", "crit-2", "crit-3", "crit-4"],
                peerIds: ["peer-1", "peer-2", "peer-3", "peer-4"]
            ),
            Assessment(
                id: "assess-3",
                title: "Evaluación Midterm",
                courseId: "course-2",
                deadline: "2025-05-01",
                criteriaIds: ["crit-1", "crit-2", "crit-3", "crit-4"],
                peerIds: ["peer-5", "peer-6"]
            ),
        ]
    }

    public func fetchCriteria() -> [Criteria] {
        Criteria.standard
    }

    public func fetchPeers() -> [Peer] {
        [
            Peer(id: "peer-1", name: "María García", initials: "MG"),
            Peer(id: "peer-2", name: "Carlos López", initials: "CL"),
            Peer(id: "peer-3", name: "Ana Martínez", initials: "AM"),
            Peer(id: "peer-4", name: "Julián Torres", initials: "JT"),
            Peer(id: "peer-5", name: "Laura Ríos", initials: "LR"),
            Peer(id: "peer-6", name: "Diego Vargas", initials: "DV"),
        ]
    }

    public func fetchEvaluations(for assessmentId: String) -> [Evaluation] {
        guard assessmentId == "assess-1" else { return [] }
        return [
            Evaluation(
                id: "eval-1",
                assessmentId: "assess-1",
                evaluatorId: "peer-1",
                peerId: "peer-2",
                scores: ["crit-1": 4, "crit-2": 5, "crit-3": 4, "crit-4": 5]
            ),
            Evaluation(
                id: "eval-2",
                assessmentId: "assess-1",
                evaluatorId: "peer-3",
                peerId: "peer-2",
                scores: ["crit-1": 3, "crit-2": 4, "crit-3": 5, "crit-4": 4]
            ),
            Evaluation(
                id: "eval-3",
                assessmentId: "assess-1",
                evaluatorId: "peer-4",
                peerId: "peer-2",
                scores: ["crit-1": 5, "crit-2": 5, "crit-3": 4, "crit-4": 5]
            ),
            Evaluation(
                id: "eval-4",
                assessmentId: "assess-1",
                evaluatorId: "peer-2",
                peerId: "peer-1",
                scores: ["crit-1": 4, "crit-2": 3, "crit-3": 4, "crit-4": 4]
            ),
            Evaluation(
                id: "eval-5",
                assessmentId: "assess-1",
                evaluatorId: "peer-3",
                peerId: "peer-1",
                scores: ["crit-1": 5, "crit-2": 4, "crit-3": 3, "crit-4": 4]
            ),
        ]
    }
}
